xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -519,4 +523,25 @@ public String getCipherSuiteName() {
     public final Locale getLanguage() {
         return language;
     }
+
+    /**
+     * Retrieves Software Version data. This method gives access to temporary Software Version data only. 
+     * @param key a <code>String</code> value of stored data ID.
+     * @return a <code>Map<String,String></code> value of data .
+     */
+    @Override
+    public Map<String, String> getSoftwareVersion() {
+return softwareVersionData; 
+    }
```

Comment:
Exposing the Map exposes implementation details. It should be possible to swap out such implementation details with affecting the API (e.g. the XEP-0092 POJO I mentioned earlier). 

IMHO, as a rule of thumb, it's almost never appropriate to expose Maps via accessor methods.